lib/domain/service/auth.dart

```diff
@@ -869,4 +881,30 @@ class AuthService extends DisposableService {
             .isBefore(PreciseDateTime.now().toUtc()) ??
         false;
   }
+
+  /// Awaits a "refreshSession" operation from the [_lockProvider] to be `null`
+  /// or old enough to upsert its own.
+  Future<void> _acquireLock() async {
+    bool acquired = false;
+
+    while (!acquired) {
+      await _lockProvider.txn(() async {
+        final at = await _lockProvider.read('refreshSession');
```

Comment:
@SleepySquash it makes sense to use something like `refreshSession($userId)` and do not contend different users on this.